backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -282,9 +322,14 @@ void cuda_drop(void *ptr, uint32_t gpu_index) {
   check_cuda_error(cudaFree(ptr));
 }
 
-/// Drop a cuda array asynchronously, if supported on the device
-void cuda_drop_async(void *ptr, cudaStream_t stream, uint32_t gpu_index) {
+/// Drop a cuda array asynchronously, if the data was allocated & it's supported
+/// on the device
+void cuda_drop_with_size_tracking_async(void *ptr, cudaStream_t stream,
```

Comment:
Probably not necessary, but have you considered adding the pointer to `size_track` here so we can assert it finishes computation as zero? This would point out memory leaks.

Of course, we can check them using compute-sanitizer as well.

Reply:
That's a good idea, I'll open an issue for that because it would require extensive modifications to this PR. I'm not sure it's easily doable because it would require to keep track for every pointer of its own size, which we don't do at the moment: we only track the memory allocated for the whole function.